wallet/psbt_manager.go

```diff
@@ -710,3 +807,717 @@ func (w *Wallet) createTxIntent(intent *FundIntent) *TxIntent {
+		// unambiguous signing path. This is critical to ensure that we
+		// have an unambiguous signing path. Our policy enforces that
+		// an input should not contain conflicting Taproot and BIP32
+		// derivation paths, nor multiple paths of the same type. This
+		// prevents misinterpretations and ensures deterministic
+		// signing.
+		isTaproot, err := validateDerivation(pInput, i)
+		if err != nil {
+			return nil, err
+		}
+
+		// Based on the validated derivation information, we dispatch
+		// the signing task to the appropriate helper function. If the
+		// input is identified as Taproot, we use
+		// `signTaprootPsbtInput`; otherwise, we assume it's a legacy
+		// or SegWit v0 input and use `signBip32PsbtInput`.
+		if isTaproot {
+			err = w.signTaprootPsbtInput(
+				ctx, packet, i, sigHashes,
+				params.InputTweakers[i],
+			)
+		} else {
+			err = w.signBip32PsbtInput(
+				ctx, packet, i, sigHashes,
+				params.InputTweakers[i],
+			)
+		}
+
+		// If an error occurred during signing, we first check if it's
+		// an error that permits us to skip the current input (e.g., if
+		// the key is not found, implying it's another signer's input
+		// in a collaborative PSBT). If the error is *not* skippable,
+		// it indicates a critical issue, and we return it immediately.
+		// Otherwise, we continue to the next input.
+		if err != nil {
+			if shouldSkipSigningError(err, i) {
+				continue
+			}
+
+			return nil, fmt.Errorf("input %d: %w", i, err)
+		}
+
+		// If signing was successful (or the error was gracefully
+		// skipped), we record the index of this input as one that the
+		// wallet has contributed a signature to. This provides
+		// valuable feedback to the caller about the progress of the
+		// signing operation.
+		signedInputs = append(signedInputs, uint32(i)) //nolint:gosec
+	}
+
+	// Finally, return the result, which includes the list of inputs that
+	// were successfully signed and the modified (partially) signed PSBT
+	// packet.
+	return &SignPsbtResult{
+		SignedInputs: signedInputs,
+		Packet:       packet,
+	}, nil
+}
+
+// parseBip32Path parses a raw BIP32 derivation path (slice of uint32) into a
+// strongly-typed structure containing the key scope and specific derivation
+// components (Account, Branch, Index).
+//
+// It enforces:
+//  1. Path length must be exactly 5.
+//  2. First 3 elements must be hardened.
+//  3. Coin type must match the wallet's chain parameters.
+//
+// Returns `ErrInvalidBip32PathLength` if the path length is not 5.
+func (w *Wallet) parseBip32Path(path []uint32) (BIP32Path, error) {
+	// The BIP32 path must have exactly 5 elements:
+	// m / purpose' / coin_type' / account' / branch / index
+	if len(path) != BIP32PathLength {
+		return BIP32Path{}, fmt.Errorf("%w: %d",
+			ErrInvalidBip32PathLength, len(path))
+	}
+
+	// The first 3 elements (Purpose, CoinType, Account) must be hardened.
+	// We check this by verifying they are >= HardenedKeyStart.
+	for i := range BIP32PathLength - 2 {
```

Comment:
Q: Why not just `3`?
```suggestion
	for i := range 3 {
```

Reply:
switched